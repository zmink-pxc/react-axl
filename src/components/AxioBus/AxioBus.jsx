import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import styles from './AxioBus.module.scss';
import {StaggeredMotion,spring} from 'react-motion';
import { AxioBusScale } from '../Utils/Scale';
const Modules = require('../Modules/load-modules.js').default;
const Controllers = require('../Controllers/load-controllers.js').default;

const Devices = Object.assign({},Modules,Controllers);

// function requireAll(r) {
//      r.keys().forEach((key)=>{console.log('New key: ' + key)})
// }

// const context = requireAll(require.context('../Modules/', true, /\.js$/));

//console.log('modules: ' + JSON.stringify(modules));

//const deviceDictionary = Object.assign({},modules,controllers);

export default class AxioBus extends React.Component {
    constructor(props){
        super(props);

        //this.deviceDictionary = deviceDictionary;

        this.defaultStyles = this.props.busConfiguration.map((key)=>{
            return {y:1000}
        })
    }

    

    render(){
        let bus = this.props.busConfiguration.map((key)=>{
            const Device = Devices[key];
            return <Device key={key}/>
        });

        const containerClass = classNames([styles.base],{[styles.wrap]:this.props.wrap})

        return (
                <div className={containerClass}>
                {/* <StaggeredMotion
                defaultStyles={this.defaultStyles}
                styles={prevInterpolatedStyles => prevInterpolatedStyles.map((_, i) => {
                    return i === 0
                    ? {y: spring(0)}
                    : {y: spring(prevInterpolatedStyles[i - 1].y)}
                })}
                >
                {interpolatingStyles =>
                <div className={styles.base}>
                {interpolatingStyles.map((style, i) => {
                    const Device = Devices[this.props.busConfiguration[i]];
                    return <Device key={i} style={{top: style.y}} />})
                }
                </div>
                }
                </StaggeredMotion> */}
                <AxioBusScale>
                {bus} 
                </AxioBusScale>
                 
                </div>            
        )
    }
}


AxioBus.propTypes = {
    busConfiguration:     PropTypes.arrayOf(PropTypes.string),
    height:               PropTypes.any,
    width:                PropTypes.any,
    partHeight:           PropTypes.any,
    wrap:                 PropTypes.bool
}

AxioBus.defaultProps = {
    wrap: false
}