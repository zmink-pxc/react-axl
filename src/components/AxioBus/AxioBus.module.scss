.base {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    flex-wrap: nowrap;
    object-fit: contain;
    height: 100%;
    width: 100%;
}

.wrap {
    flex-wrap: wrap;
}

.flexCenter {
    justify-content: center;
}