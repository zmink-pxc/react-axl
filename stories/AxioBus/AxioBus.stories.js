import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import AxioBus from '../../src/components/AxioBus/AxioBus.jsx';

var isHover = false;
var pos = '0px';

storiesOf('AxioBus', module)
  .addDecorator(withKnobs)
  .add('Basic', () => <AxioBus busConfiguration={['2404267','2701916','2688310','2702525']}/>)
  .add('Scaling', () => <div style={{width: "400px",height: "500px",backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267']}/></div>)
  .add('Scaling2', () => <div style={{width: "400px",height: "500px",backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','2688310','2702525']}/></div>)
  .add('Scaling with Padding', () => <div style={{width: "400px",height: "500px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','2688310','2702525']}/></div>)
  .add('Bus props', () => <div style={{width: "400px",height: "500px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','2688310','2702525','1088127','1088129','1088062','1088123']} busProps={[{style:{backgroundColor:"red"}},null,null,null,{style:{backgroundColor:"red"}}]}/></div>)
  .add('Basic with Smart Elements', () => <div style={{width: "400px",height: "500px",boxSizing: 'border-box',padding:'40px 40px',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','2688310','2702071','1088127','1088129','1088062','1088123']} /></div>)
  .add('Scaling with Smart Elements', () => <div style={{width: "1500px",height: "312px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','1088127','1088129','1088062','1088123']}/></div>)
  .add('SCA Rendering 1', () => <div style={{width: "1500px",height: "312px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','1088127','1088123','1088062']}/></div>)
  .add('SCA Rendering 2', () => <div style={{width: "1500px",height: "312px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','1088127','1088123']}/></div>)
  .add('SCA Rendering 3', () => <div style={{width: "1500px",height: "312px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','1088127','1088123','1088127','1088123','1088123']}/></div>)
  .add('Uknown Card', () => <div style={{width: "1500px",height: "312px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','1088127','1088123','1088127','1088123','xdfasdf']}/></div>);

