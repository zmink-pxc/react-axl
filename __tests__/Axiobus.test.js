import React from 'react';
import AxioBus from '@components/AxioBus/AxioBus.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    var component = renderer.create(<AxioBus busConfiguration={['2404267','2701916','2688310','2702525']}/>)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(<div style={{width: "400px",height: "500px",backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267']}/></div>);

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();


    component = renderer.create(<div style={{width: "1500px",height: "312px",boxSizing: 'border-box',padding:'40px 40px',backgroundColor: 'orange',position: 'absolute',left:'50px',top: '100px'}}><AxioBus busConfiguration={['2404267','2701916','1088127','1088129','1088062','1088123']}/></div>);

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});