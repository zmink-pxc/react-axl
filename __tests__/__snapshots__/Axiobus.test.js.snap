// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="base"
  style={Object {}}
/>
`;

exports[`renders correctly 2`] = `
<div
  style={
    Object {
      "backgroundColor": "orange",
      "height": "500px",
      "left": "50px",
      "position": "absolute",
      "top": "100px",
      "width": "400px",
    }
  }
>
  <div
    className="base"
    style={Object {}}
  />
</div>
`;

exports[`renders correctly 3`] = `
<div
  style={
    Object {
      "backgroundColor": "orange",
      "boxSizing": "border-box",
      "height": "312px",
      "left": "50px",
      "padding": "40px 40px",
      "position": "absolute",
      "top": "100px",
      "width": "1500px",
    }
  }
>
  <div
    className="base"
    style={Object {}}
  />
</div>
`;
